pandas/_libs/tslibs/period.pyx

```diff
@@ -1186,6 +1186,13 @@ cdef int64_t period_ordinal_to_dt64(int64_t ordinal, int freq) except? -1:
     if ordinal == NPY_NAT:
         return NPY_NAT
 
+    if freq == 11000:
+        # Microsecond, avoid get_date_info to prevent floating point errors
+        pandas_datetime_to_datetimestruct(ordinal, NPY_FR_us, &dts)
```

Comment:
if this path is kept, then I would do if/else here and have: L 1192, L1194 after (as they are repeated at L1197, 1198)

Reply:
will do

Reply:
k looks fine, what about higher freq than microsecond? these are ok in the get_date_info path?

Reply:
> what about higher freq than microsecond? these are ok in the get_date_info path?

I've got another branch im working on to handle that; it'll end up de-duplicating some code in the process

Reply:
kk